File: warmUps/ruby_wmup/scrabble/scrabble.js
function scrabble(word){
  scr = 0;

  var p1 = word.match(/[AEIOULNRST]/gi);
  var p2 = word.match(/[DG]/gi);
  var p3 = word.match(/[BCMP]/gi);
  var p4 = word.match(/[FHVWY]/gi);
  var p5 = word.match(/[K]/gi);
  var p6 = word.match(/[JX]/gi);
  var p7 = word.match(/[QZ]/gi);

  if(p1!==null){
    scr += p1.length;
  }
  if(p2!==null){
    scr += p2.length*2;
  }
  if(p3!==null){
    scr += p3.length*3;
  }
  if(p4!==null){
    scr += p4.length*4;
  }
  if(p5!==null){
    scr += p5.length*5;
  }
  if(p6!==null){
    scr += p6.length*8;
  }
  if(p7!==null){
    scr += p6.length*10;
  }
  return scr;
}

console.log(scrabble('abc'));
console.log(scrabble('kf'));
console.log(scrabble('qx'));

File: warmUps/ruby_wmup/scrabble/scrabble.rb
require 'pry'

class Scrabble
  attr_accessor :score
  def initialize(word)
    @score = calScrab(word)
  end

  def self.calScrab(str)
    scr = 0;
    probe1 = str.scan(/[AEIOULNRST]/i)#23458 10
    probe2 = str.scan(/[DG]/i)
    probe3 = str.scan(/[BCMP]/i)
    probe4 = str.scan(/[FHVWY]/i)
    probe5 = str.scan(/[K]/i)
    probe6 = str.scan(/[JX]/i)
    probe7 = str.scan(/[QZ]/i)
    if !probe1.empty?
      scr += probe1.length
    end
    if !probe2.empty?
      scr += probe2.length*2
    end
    if !probe3.empty?
      scr += probe3.length*3
    end
    if !probe4.empty?
      scr += probe4.length*4
    end
    if !probe5.empty?
      scr += probe5.length*5
    end
    if !probe6.empty?
      scr += probe6.length*8
    end
    if !probe7.empty?
      scr += probe7.length*10
    end
    scr
  end

end

binding.pry
